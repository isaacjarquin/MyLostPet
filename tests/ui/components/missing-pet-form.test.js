import "react-native"
import React from "react"
import MissingPetForm from "../../../components/missing-pet-form"
import { shallow } from "enzyme"
import toJSON from "enzyme-to-json"

describe("MissingPetForm", () => {
    const navigate = jest.fn()

    const props = {
        navigation: {
            navigate: navigate
        }
    }

    beforeEach(() => {
        wrapper = shallow(<MissingPetForm {...props} />)
    })

    afterEach(() => {
        navigate.mockReset()
    })

    it("renders correctly", () => {
        expect(wrapper).toBeDefined()
    })
})
