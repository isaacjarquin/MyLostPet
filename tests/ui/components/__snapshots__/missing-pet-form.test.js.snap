// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissingPetForm renders correctly 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  keyboardVerticalOffset={100}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollViewMock>
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          backgroundColor="grey"
          color="white"
          name="person"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={30}
          style={
            Object {
              "borderColor": "grey",
              "borderWidth": 0.5,
              "marginBottom": 4,
              "marginLeft": 4,
              "marginTop": 4,
              "paddingLeft": 4,
              "paddingTop": 8,
              "width": 40,
            }
          }
          type="MaterialIcons"
          underlayColor="white"
        />
        <TextInput
          allowFontScaling={true}
          borderColor="grey"
          onChangeText={[Function]}
          placeholder="Introduzca aquí su nombre"
          placeholderTextColor="grey"
          style={
            Array [
              Object {
                "borderColor": "grey",
                "borderWidth": 0.5,
                "color": "grey",
                "fontSize": 14,
                "height": 50,
                "marginBottom": 4,
                "marginRight": 4,
                "marginTop": 4,
                "padding": 10,
                "paddingLeft": 15,
                "width": "87%",
              },
              Object {
                "color": "grey",
              },
            ]
          }
          underlineColorAndroid="white"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          backgroundColor="grey"
          color="white"
          name="email"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={30}
          style={
            Object {
              "borderColor": "grey",
              "borderWidth": 0.5,
              "marginBottom": 4,
              "marginLeft": 4,
              "marginTop": 4,
              "paddingLeft": 4,
              "paddingTop": 8,
              "width": 40,
            }
          }
          type="Entypo"
          underlayColor="white"
        />
        <TextInput
          allowFontScaling={true}
          borderColor="grey"
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Introduzca aquí su correo electronico"
          placeholderTextColor="grey"
          style={
            Array [
              Object {
                "borderColor": "grey",
                "borderWidth": 0.5,
                "color": "grey",
                "fontSize": 14,
                "height": 50,
                "marginBottom": 4,
                "marginRight": 4,
                "marginTop": 4,
                "padding": 10,
                "paddingLeft": 15,
                "width": "87%",
              },
              Object {
                "color": "grey",
              },
            ]
          }
          underlineColorAndroid="white"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          backgroundColor="grey"
          color="white"
          name="pets"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={30}
          style={
            Object {
              "borderColor": "grey",
              "borderWidth": 0.5,
              "marginBottom": 4,
              "marginLeft": 4,
              "marginTop": 4,
              "paddingLeft": 4,
              "paddingTop": 8,
              "width": 40,
            }
          }
          type="MaterialIcons"
          underlayColor="white"
        />
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "#d6d7da",
                "borderWidth": 0.5,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginRight": 4,
                "marginTop": 4,
                "padding": 15,
                "width": "87%",
              },
              Object {
                "borderColor": "grey",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "width": "95%",
                },
                Object {
                  "color": "grey",
                },
              ]
            }
          >
            Seleccione el tipo de mascota
          </Text>
          <Icon
            color="grey"
            name="keyboard-arrow-down"
            raised={false}
            reverse={false}
            reverseColor="white"
            size={20}
            type="MaterialIcons"
            underlayColor="white"
          />
        </TouchableOpacity>
        <SecondaryMenuModal
          handler={[Function]}
          hidePetTypeModal={[Function]}
          isVisible={false}
          items={
            Array [
              Object {
                "id": 1,
                "value": "perro",
              },
              Object {
                "id": 2,
                "value": "gato",
              },
              Object {
                "id": 3,
                "value": "conejo",
              },
              Object {
                "id": 4,
                "value": "hamster",
              },
              Object {
                "id": 5,
                "value": "iguana",
              },
              Object {
                "id": 6,
                "value": "hurón",
              },
              Object {
                "id": 7,
                "value": "tortuga",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          backgroundColor="grey"
          color="white"
          name="pets"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={30}
          style={
            Object {
              "borderColor": "grey",
              "borderWidth": 0.5,
              "marginBottom": 4,
              "marginLeft": 4,
              "marginTop": 4,
              "paddingLeft": 4,
              "paddingTop": 8,
              "width": 40,
            }
          }
          type="MaterialIcons"
          underlayColor="white"
        />
        <TextInput
          allowFontScaling={true}
          borderColor="grey"
          onChangeText={[Function]}
          placeholder="Introduzca aquí la raza del animal encontrado"
          placeholderTextColor="grey"
          style={
            Array [
              Object {
                "borderColor": "grey",
                "borderWidth": 0.5,
                "color": "grey",
                "fontSize": 14,
                "height": 50,
                "marginBottom": 4,
                "marginRight": 4,
                "marginTop": 4,
                "padding": 10,
                "paddingLeft": 15,
                "width": "87%",
              },
              Object {
                "color": "grey",
              },
            ]
          }
          underlineColorAndroid="white"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          backgroundColor="grey"
          color="white"
          name="ruler"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={30}
          style={
            Object {
              "borderColor": "grey",
              "borderWidth": 0.5,
              "marginBottom": 4,
              "marginLeft": 4,
              "marginTop": 4,
              "paddingLeft": 4,
              "paddingTop": 8,
              "width": 40,
            }
          }
          type="entypo"
          underlayColor="white"
        />
        <TextInput
          allowFontScaling={true}
          borderColor="grey"
          onChangeText={[Function]}
          placeholder="Introduzca aquí el tamaño aproximado"
          placeholderTextColor="grey"
          style={
            Array [
              Object {
                "borderColor": "grey",
                "borderWidth": 0.5,
                "color": "grey",
                "fontSize": 14,
                "height": 50,
                "marginBottom": 4,
                "marginRight": 4,
                "marginTop": 4,
                "padding": 10,
                "paddingLeft": 15,
                "width": "87%",
              },
              Object {
                "color": "grey",
              },
            ]
          }
          underlineColorAndroid="white"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          backgroundColor="grey"
          color="white"
          name="calendar"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={30}
          style={
            Object {
              "borderColor": "grey",
              "borderWidth": 0.5,
              "marginBottom": 4,
              "marginLeft": 4,
              "marginTop": 4,
              "paddingLeft": 4,
              "paddingTop": 8,
              "width": 40,
            }
          }
          type="evilicon"
          underlayColor="white"
        />
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "borderWidth": 0.5,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                  "marginRight": 4,
                  "marginTop": 4,
                  "paddingRight": 5,
                  "width": 702,
                },
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "grey",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "marginBottom": 10,
                    "marginLeft": 15,
                    "marginTop": 10,
                    "paddingBottom": 5,
                    "paddingTop": 5,
                  },
                  Object {
                    "color": "grey",
                  },
                ]
              }
            >
              Seleccione cuando encontro a la mascota
            </Text>
            <Icon
              color="grey"
              name="select-arrows"
              raised={false}
              reverse={false}
              reverseColor="white"
              size={20}
              type="entypo"
              underlayColor="white"
            />
          </View>
        </TouchableOpacity>
        <CustomDatePickerIOS
          cancelTextIOS="Cancel"
          confirmTextIOS="Confirm"
          date={2017-12-27T19:05:11.937Z}
          isVisible={false}
          mode="date"
          neverDisableConfirmIOS={false}
          onCancel={[Function]}
          onConfirm={[Function]}
          onHideAfterConfirm={[Function]}
          reactNativeModalPropsIOS={Object {}}
          titleIOS="Pick a date"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          backgroundColor="grey"
          color="white"
          name="place"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={30}
          style={
            Object {
              "borderColor": "grey",
              "borderWidth": 0.5,
              "marginBottom": 4,
              "marginLeft": 4,
              "marginTop": 4,
              "paddingLeft": 4,
              "paddingTop": 8,
              "width": 40,
            }
          }
          type="MaterialIcons"
          underlayColor="white"
        />
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "#d6d7da",
                "borderWidth": 0.5,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginRight": 4,
                "marginTop": 4,
                "padding": 15,
                "width": "87%",
              },
              Object {
                "borderColor": "grey",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "width": "95%",
                },
                Object {
                  "color": "grey",
                },
              ]
            }
          >
            Comunidad autónoma
          </Text>
          <Icon
            color="grey"
            name="keyboard-arrow-down"
            raised={false}
            reverse={false}
            reverseColor="white"
            size={20}
            type="MaterialIcons"
            underlayColor="white"
          />
        </TouchableOpacity>
        <SecondaryMenuModal
          handler={[Function]}
          hidePetTypeModal={[Function]}
          isVisible={false}
          items={
            Array [
              Object {
                "id": "andalucia",
                "provincias": Array [
                  Object {
                    "id": 1,
                    "value": "Almería",
                  },
                  Object {
                    "id": 2,
                    "value": "Cádiz",
                  },
                  Object {
                    "id": 3,
                    "value": "Córdoba",
                  },
                  Object {
                    "id": 4,
                    "value": "Granada",
                  },
                  Object {
                    "id": 5,
                    "value": "Huelva",
                  },
                  Object {
                    "id": 6,
                    "value": "Jaen",
                  },
                  Object {
                    "id": 7,
                    "value": "Málaga",
                  },
                  Object {
                    "id": 8,
                    "value": "Sevilla",
                  },
                ],
                "value": "Andalucia",
              },
              Object {
                "id": "aragon",
                "provincias": Array [
                  Object {
                    "id": 9,
                    "value": "Huesca",
                  },
                  Object {
                    "id": 10,
                    "value": "Teruel",
                  },
                  Object {
                    "id": 11,
                    "value": "Zaragoza",
                  },
                ],
                "value": "Aragon",
              },
              Object {
                "id": "canarias",
                "provincias": Array [
                  Object {
                    "id": 12,
                    "value": "Las Palmas",
                  },
                  Object {
                    "id": 13,
                    "value": "Santa Cruz de Tenerife",
                  },
                ],
                "value": "Canarias",
              },
              Object {
                "id": "cantabria",
                "provincias": Array [
                  Object {
                    "id": 14,
                    "value": "Cantabria",
                  },
                ],
                "value": "Cantabria",
              },
              Object {
                "id": "castilla_la_mancha",
                "provincias": Array [
                  Object {
                    "id": 15,
                    "value": "Albacete",
                  },
                  Object {
                    "id": 16,
                    "value": "Ciudad Real",
                  },
                  Object {
                    "id": 17,
                    "value": "Cuenca",
                  },
                  Object {
                    "id": 18,
                    "value": "Guadalajara",
                  },
                  Object {
                    "id": 19,
                    "value": "Toledo",
                  },
                ],
                "value": "Castilla la Mancha",
              },
              Object {
                "id": "castilla_y_leon",
                "provincias": Array [
                  Object {
                    "id": 20,
                    "value": "Ávila",
                  },
                  Object {
                    "id": 21,
                    "value": "Burgos",
                  },
                  Object {
                    "id": 22,
                    "value": "Leon",
                  },
                  Object {
                    "id": 23,
                    "value": "Palencia",
                  },
                  Object {
                    "id": 24,
                    "value": "Salamanca",
                  },
                  Object {
                    "id": 25,
                    "value": "Segovia",
                  },
                  Object {
                    "id": 26,
                    "value": "Soria",
                  },
                  Object {
                    "id": 27,
                    "value": "Valladolid",
                  },
                  Object {
                    "id": 28,
                    "value": "Zamora",
                  },
                ],
                "value": "Castilla y Leon",
              },
              Object {
                "id": "cataluna",
                "provincias": Array [
                  Object {
                    "id": 29,
                    "value": "Barcelona",
                  },
                  Object {
                    "id": 30,
                    "value": "Gerona",
                  },
                  Object {
                    "id": 31,
                    "value": "Lerida",
                  },
                  Object {
                    "id": 32,
                    "value": "Tarragona",
                  },
                ],
                "value": "Cataluña",
              },
              Object {
                "id": "ceuta",
                "provincias": Array [
                  Object {
                    "id": 33,
                    "value": "Ceuta",
                  },
                ],
                "value": "Ceuta",
              },
              Object {
                "id": "valencia",
                "provincias": Array [
                  Object {
                    "id": 34,
                    "value": "Alicante",
                  },
                  Object {
                    "id": 35,
                    "value": "Castellón",
                  },
                  Object {
                    "id": 36,
                    "value": "Valencia",
                  },
                ],
                "value": "Comunidad Valenciana",
              },
              Object {
                "id": "extremadura",
                "provincias": Array [
                  Object {
                    "id": 37,
                    "value": "Badajoz",
                  },
                  Object {
                    "id": 38,
                    "value": "Cáceres",
                  },
                ],
                "value": "Extremadura",
              },
              Object {
                "id": "baleares",
                "provincias": Array [
                  Object {
                    "id": 39,
                    "value": "Islas Baleares",
                  },
                ],
                "value": "Islas Baleares",
              },
              Object {
                "id": "galicia",
                "provincias": Array [
                  Object {
                    "id": 40,
                    "value": "La Coruña",
                  },
                  Object {
                    "id": 41,
                    "value": "Lugo",
                  },
                  Object {
                    "id": 42,
                    "value": "Orense",
                  },
                  Object {
                    "id": 43,
                    "value": "Pontevedra",
                  },
                ],
                "value": "Galicia",
              },
              Object {
                "id": "madrid",
                "provincias": Array [
                  Object {
                    "id": 44,
                    "value": "Comunidad de Madrid",
                  },
                ],
                "value": "Madrid",
              },
              Object {
                "id": "melilla",
                "provincias": Array [
                  Object {
                    "id": 45,
                    "value": "Melilla",
                  },
                ],
                "value": "Melilla",
              },
              Object {
                "id": "murcia",
                "provincias": Array [
                  Object {
                    "id": 46,
                    "value": "Región de Murcia",
                  },
                ],
                "value": "Murcia",
              },
              Object {
                "id": "navarra",
                "provincias": Array [
                  Object {
                    "id": 47,
                    "value": "Comunidad foral de Navarra",
                  },
                ],
                "value": "Navarra",
              },
              Object {
                "id": "pais_vasco",
                "provincias": Array [
                  Object {
                    "id": 48,
                    "value": "Álava",
                  },
                  Object {
                    "id": 49,
                    "value": "Gipúzcoa",
                  },
                  Object {
                    "id": 50,
                    "value": "Vizcaya",
                  },
                ],
                "value": "País Vasco",
              },
              Object {
                "id": "asturias",
                "provincias": Array [
                  Object {
                    "id": 51,
                    "value": "Principado de Asturias",
                  },
                ],
                "value": "Principado de asturias",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          backgroundColor="grey"
          color="white"
          name="place"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={30}
          style={
            Object {
              "borderColor": "grey",
              "borderWidth": 0.5,
              "marginBottom": 4,
              "marginLeft": 4,
              "marginTop": 4,
              "paddingLeft": 4,
              "paddingTop": 8,
              "width": 40,
            }
          }
          type="MaterialIcons"
          underlayColor="white"
        />
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderColor": "#d6d7da",
                "borderWidth": 0.5,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginRight": 4,
                "marginTop": 4,
                "padding": 15,
                "width": "87%",
              },
              Object {
                "borderColor": "grey",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "width": "95%",
                },
                Object {
                  "color": "grey",
                },
              ]
            }
          >
            Provincia
          </Text>
          <Icon
            color="grey"
            name="keyboard-arrow-down"
            raised={false}
            reverse={false}
            reverseColor="white"
            size={20}
            type="MaterialIcons"
            underlayColor="white"
          />
        </TouchableOpacity>
        <SecondaryMenuModal
          handler={[Function]}
          hidePetTypeModal={[Function]}
          isVisible={false}
          items={Array []}
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          backgroundColor="grey"
          color="white"
          name="place"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={30}
          style={
            Object {
              "borderColor": "grey",
              "borderWidth": 0.5,
              "marginBottom": 4,
              "marginLeft": 4,
              "marginTop": 4,
              "paddingLeft": 4,
              "paddingTop": 8,
              "width": 40,
            }
          }
          type="MaterialIcons"
          underlayColor="white"
        />
        <TextInput
          allowFontScaling={true}
          borderColor="grey"
          onChangeText={[Function]}
          placeholder="Introduzca aquí la ciudad o municipio"
          placeholderTextColor="grey"
          style={
            Array [
              Object {
                "borderColor": "grey",
                "borderWidth": 0.5,
                "color": "grey",
                "fontSize": 14,
                "height": 50,
                "marginBottom": 4,
                "marginRight": 4,
                "marginTop": 4,
                "padding": 10,
                "paddingLeft": 15,
                "width": "87%",
              },
              Object {
                "color": "grey",
              },
            ]
          }
          underlineColorAndroid="white"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Icon
          backgroundColor="grey"
          color="white"
          name="text"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={30}
          style={
            Object {
              "borderColor": "grey",
              "borderWidth": 0.5,
              "marginBottom": 4,
              "marginLeft": 4,
              "marginTop": 4,
              "paddingLeft": 4,
              "paddingTop": 8,
              "width": 40,
            }
          }
          type="entypo"
          underlayColor="white"
        />
        <TextInput
          allowFontScaling={true}
          borderColor="grey"
          onChangeText={[Function]}
          placeholder="Introduzca aquí una description del animal"
          placeholderTextColor="grey"
          style={
            Array [
              Object {
                "borderColor": "grey",
                "borderWidth": 0.5,
                "color": "grey",
                "fontSize": 14,
                "height": 50,
                "marginBottom": 4,
                "marginRight": 4,
                "marginTop": 4,
                "padding": 10,
                "paddingLeft": 15,
                "width": "87%",
              },
              Object {
                "color": "grey",
              },
            ]
          }
          underlineColorAndroid="white"
          value=""
        />
      </View>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "grey",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "margin": 4,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#99d1ed",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "marginLeft": 20,
                "marginRight": 10,
                "paddingTop": 10,
              }
            }
          >
            Añade una foto de la mascota
          </Text>
          <Icon
            color="white"
            name="plus"
            raised={false}
            reverse={false}
            reverseColor="white"
            size={50}
            type="evilicon"
            underlayColor="white"
          />
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#333333",
            "opacity": 0.8,
            "padding": 25,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
              "fontSize": 16,
            }
          }
        >
          Guardar datos
        </Text>
      </TouchableOpacity>
    </View>
  </ScrollViewMock>
  <SafeAreaView />
</KeyboardAvoidingView>
`;
